app/services/vacancy_filters.rb

```diff
@@ -30,9 +30,13 @@ def to_hash
     }
   end
 
+  def only_active_to_hash
+    to_hash.delete_if { |_, v| v.blank? || v.nil? }
+  end
+
   def any?
-    filters = to_hash
-    filters.delete_if { |k, v| k.eql?(:radius) || v.nil? }
+    filters = only_active_to_hash
+    filters.delete_if { |k, _| k.eql?(:radius) }
```

Comment:
This `k.eql?(:radius)` seems a bit redundant now, as all we're doing is casting the radius to a string. It might be better to just cast the radius to a string if it's present in the initializer. Something like `radius = args[:radius].to_s if args[:radius].present?`